// visa-sdk/dist/Visa.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fs = require("fs");
const rest = require("./utils/rest");
var Endpoint;
(function (Endpoint) {
    Endpoint["Sandbox"] = "https://sandbox.api.visa.com";
    Endpoint["Certification"] = "https://cert.api.visa.com";
    Endpoint["Production"] = "https://api.visa.com";
})(Endpoint = exports.Endpoint || (exports.Endpoint = {}));
class Visa {
    constructor({ userId, password, certificate_passphrase, certificate_file_path }, host = Endpoint.Sandbox) {
        const certificate = fs.readFileSync(certificate_file_path);
        this._host = host;
        this._credentials = {
            userId,
            certificate,
            password,
            certificate_passphrase
        };
    }
    get(uri, options) {
        return rest._get(this._credentials, `${this._host}${uri}`, options);
    }
    delete(uri, options) {
        return rest._delete(this._credentials, `${this._host}${uri}`, options);
    }
    put(uri, body, options) {
        return rest._put(this._credentials, `${this._host}${uri}`, body, options);
    }
    post(uri, body, options) {
        return rest._post(this._credentials, `${this._host}${uri}`, body, options);
    }
    helloWorld() {
        return this.get(`/vdp/helloworld`);
    }
}
exports.Visa = Visa;
