// visa-sdk/dist/VPA.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const Visa_1 = require("./Visa");
class VPA extends Visa_1.Visa {
    virtualCardRequisition(requisition) {
        return this.post(`/vpa/v1/accountManagement/VirtualCardRequisition`, requisition);
    }
    managePaymentControls(parameters) {
        return this.post(`/vpa/v1/accountManagement/ManagePaymentControls`, parameters);
    }
    listPaymentControl(parameters) {
        return this.post(`/vpa/v1/accountManagement/ListPaymentControl`, parameters);
    }
    getAccountStatus(query) {
        return this.post(`/vpa/v1/accountManagement/GetAccountStatus`, query);
    }
    getSecurityCode(parameters) {
        return this.post(`/vpa/v1/accountManagement/GetSecurityCode`, parameters);
    }
    requestVirtualAccount(parameters) {
        return this.post(`/vpa/v1/requisitionService`, parameters);
    }
    SUAGetAccountStatus(parameters) {
        return this.post(`/vpa/v1/proxy/GetAccountStatus`, parameters);
    }
    manageProxy(parameters) {
        return this.post(`/vpa/v1/proxy/ManageProxy`, parameters);
    }
    processPayments(parameters) {
        return this.post(`/vpa/v1/payment/ProcessPayments`, parameters);
    }
    resendPayment(parameters) {
        return this.post(`/vpa/v1/payment/ResendPayment`, parameters);
    }
    getPaymentUrl(parameters) {
        return this.post(`/vpa/v1/payment/GetPaymentDetailURL`, parameters);
    }
    getPaymentDetails(parameters) {
        return this.post(`/vpa/v1/payment/GetPaymentDetails`, parameters);
    }
    cancelPayment(parameters) {
        return this.post(`/vpa/v1/payment/CancelPayment`, parameters);
    }
    createSupplier(parameters) {
        return this.post(`/vpa/v1/supplier/CreateSupplier`, parameters);
    }
    updateSupplier(parameters) {
        return this.post(`/vpa/v1/supplier/UpdateSupplier`, parameters);
    }
    disableSupplier(parameters) {
        return this.post(`/vpa/v1/supplier/DisableSupplier`, parameters);
    }
    getSupplier(parameters) {
        return this.post(`/vpa/v1/supplier/GetSupplierDetails`, parameters);
    }
    manageSupplierAccount(parameters) {
        return this.post(`/vpa/v1/supplier/ManageSupplierAccount`, parameters);
    }
}
exports.VPA = VPA;
