// visa-sdk/dist/utils/rest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const request = require("request");
const debug_1 = require("debug");
const log = debug_1.default('visa:rest');
function _get(credentials, uri, options) {
    const params = Object.assign({}, options, { uri, method: 'GET' });
    return _req(credentials, params);
}
exports._get = _get;
function _delete(credentials, uri, options) {
    const params = Object.assign({}, options, { uri, method: 'DELETE' });
    return _req(credentials, params);
}
exports._delete = _delete;
function _put(credentials, uri, body, options) {
    const params = Object.assign({}, options, { uri,
        body, method: 'PUT' });
    return _req(credentials, params);
}
exports._put = _put;
function _post(credentials, uri, body, options) {
    const params = Object.assign({}, options, { uri,
        body, method: 'POST' });
    return _req(credentials, params);
}
exports._post = _post;
function _req(credentials, options) {
    return new Promise((resolve, reject) => {
        const defaultOptions = {
            headers: {
                'Authorization': `Basic ${new Buffer(credentials.userId + ':' + credentials.password).toString('base64')}`
            },
            agentOptions: {
                pfx: credentials.certificate,
                passphrase: credentials.certificate_passphrase
            },
            json: true
        };
        defaultObjectProperties(options, defaultOptions);
        log(options);
        request(options, (error, response, body) => {
            if (error)
                return reject(error);
            return resolve({ response, body });
        });
    });
}
exports._req = _req;
function defaultObjectProperties(target, defaults) {
    for (const [key, value] of Object.entries(defaults)) {
        if (!target[key]) {
            target[key] = value;
            continue;
        }
        ;
        defaultObjectProperties(target[key], defaults[key]);
    }
}
